tests/test_ble_interface_core.py

```
@@ -411,6 +449,973 @@ def test_ble_interface_defaults_auto_reconnect_disabled(
+            exc_type: type[BaseException] | None,
+            exc: BaseException | None,
+            tb: TracebackType | None,
+        ) -> Literal[False]:
+            _ = (exc_type, exc, tb)
+            self._lock.release()
+            return False
+
+    observed_address_lock = _ObservedAddressLock()
+
+    @contextlib.contextmanager
+    def _observed_addr_lock_context(_addr: str | None):
```

Ruff (0.15.4)

[warning] Missing return type annotation for private function `_observed_addr_lock_context`

(ANN202)